internal/oci/ecr/round_tripper.go

```diff
@@ -0,0 +1,18 @@
+package ecr
+
+import "net/http"
+
+type RoundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (fn RoundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	resp, err := fn(r)
+	if err != nil {
+		return resp, err
+	}
+	// when authorization token has expired, AWS ECR response with http code 403 but oras expected
+	// http code 401 in this case.
+	if resp.StatusCode == http.StatusForbidden {
+		resp.StatusCode = http.StatusUnauthorized
+	}
```

Comment:
I think this isn't enough! You're also need to put a challange, bacause it expects one to enter in the switch case statement which will call the credentials function!

Reference: https://github.com/oras-project/oras-go/blob/6fce64e956e850a3af7ea16e85959c23e3eddc4b/registry/remote/auth/client.go#L212

Reply:
I think this should be enough, as we're using `user:pass` it should be a basic auth:
```go
resp.Header.Set("WWW-Authenticate", `Basic realm="ecr repo"`)
```

Reply:
@erka I was thinking about this and the differences between public and private ECR! Maybe we can make the `RoundTriper` smarter by storing the `www-authenticate` when the request returns `401` for the very first time

I think it's better than a naive solution like just making an if statement to compare the host to know if it's public or not

WDYT?

Reply:
@thepabloaguilar the reality is bigger a bit.

Somehow I never faced the fact that AWS has `ecr` and `ecr-public` as separated services with own api for them (sdk, cli, logins, etc..). So Flipt tries to support `private` version currently and we should align the documentation. 

ORAS does a lot of things base on `www-authenticate` scheme and I think we should not mess there. I try to find documentation about http response codes and how to provide right bridge/adaptor between AWS responses and ORAS expectations.  From observations some responses could be http 400 with message `"code":"DENIED","message":"Your Authorization Token is invalid."`

Reply:
Unfortunately the AWS documentation is not accurate (not just for ECS but for all their services) and we're not going to cover every case because of that! From my pov we should support the cases we already know that is possible to happen

Also, about the private ecr stuff, _ORAS_ don't look to anything than the auth schema when it's _Basic_

Reply:
It looks a bit hacky this way. I increasingly favor your suggestion to moving cache credentials to the ECR struct.

Reply:
Yeah, at least we don't have to mimic a behavior and already has the information about when the token expires